### ISSUE TITLE
Unable to Install for Language English/No Country

### ISSUE THREAD

**Issue by vosspl** [pop-os/installer#245]
**Distribution (run `cat /etc/os-release`):**
 Pop!_OS 21.04
 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
Installer

**Issue/Bug Description:**
I tried installing Pop!_OS in English. After selecting the language, the installer prompts you to select a country. As I'm from Germany trying to use the OS in English, I was unable to select the correct option for me (Germany) so I went with "None" instead. This lead to the Installer failing to configure the system later down the line. Selecting German/Germany instead resolved the issue,

**Steps to reproduce (if you know):**
Start installation, select English and then None, continue the installation. it should fail before prompting a restart.

**Expected behavior:**
The installation shouldn't fail.

**Other Notes:**
To me it would be preferable if I was just able to select any country, no matter what language I selected.

**Comment by jacobgkau** [comment]
Caused by https://github.com/pop-os/installer/pull/241 (although installation would have failed if `None` was selected even before that change to remove it from the list.)

It does seem like allowing any country to be selected would make sense.

**Comment by mmstick** [comment]
`en_DE` is not a valid locale. You can see available supported English localizations with `ls /usr/share/i18n/locales | grep en`. In general, `None` should be automatically converted to `US`.

**Comment by vosspl** [comment]
You're of course right about that. I restarted the process and found out my initial confusion was created by the installer's UX. When starting the installer, English is pre-selected with a tick on the right side. The button to continue installation is at the bottom right, suggesting it will take me to the next page of the installer.

That is what I expected the button to do, not for it to offer me with a selection of sub options. So while the window obviously still states "Languages" on the left, the way I interacted with the installer suggested to me that I was done selecting a language and being presented with a list of countries I assumed I was now meant to select my home country.

![Screenshot 2021-12-07 180209](https://user-images.githubusercontent.com/29821337/145073392-e587127a-0cf3-4439-99f5-e2ed49610883.jpg)

I think it would be preferable if "English" wasn't pre-selected. That way, I'd be required to click on English, which would take me to the list of options for that language. The "Select" button at the bottom shouldn't work until I selected an actual language, instead of just having selected a language Family. At the very least English shouldn't be marked with a tick as it currently is, but with an arrow, suggesting I need to select more options.

**Comment by 13r0ck** [comment]
I don't believe we built new 21.04 ISOs since #241, so this has already been fixed, we just need to rebuild the ISO's for 21.04 so users can't select none

**Comment by jacobgkau** [comment]
@13r0ck This was caused by #241, not fixed by #241. The user is requesting the "None" option that we removed.

**Comment by 13r0ck** [comment]
> Expected behavior:
The installation shouldn't fail.

> Selecting German/Germany instead resolved the issue

No clearly not.

**Comment by jacobgkau** [comment]
> As I'm from Germany trying to use the OS in English, I was unable to select the correct option for me (Germany)

> Selecting German/Germany instead resolved the issue,

This was the UX concern, which still exists. The user ended up installing with a German language just because they thought they needed to get the country correct.

**Comment by jacobgkau** [comment]
Issues are closed when their PRs are merged, not when they're released, so if you think an issue is actually resolved, close it.

Also,

> we just need to rebuild the ISO's for 21.04

We will not be building any further 21.04 iso's since 21.10 is already released.

**Comment by 13r0ck** [comment]
Neither is it *caused* by. The crash is clearly *caused* by selecting `None`. There are clearly ways to improve the installer. But how could the user have selected `None` when #241 removes `None`. The user clearly didn't have #241, so there is no way it caused the error.

**Comment by jacobgkau** [comment]
That is fair, my initial reading of the issue assumed they had #241 when they did not, upon reading closer. I read the context of the user's confusion and focused on that, and I didn't look too much closer today based on nonsensically saying we needed to re-spin isos for a non-current release. The initial description also included:

> To me it would be preferable if I was just able to select any country, no matter what language I selected.

Which is what this issue is about at this point. As Michael explained, we can't have all countries for all languages, but an "Other" option that selects US in the background might provide a release valve for users who think the country in this list matters beyond the language localization.

**Comment by 13r0ck** [comment]
Agreed, I would like an "Other" option, but it sounds like if we did that the default would just be "US", which is not the same as "Other". I think me might have to come up with another solution, or pull UX in.

**Comment by jacobgkau** [comment]
> it sounds like if we did that the default would just be "US", which is not the same as "Other".

There _is_ no "Other" or "None" locale, that's the crux of the issue. The locale needs to be `en_SOMETHING`, and only English-native countries are going to have variants of English, so one of them needs to be chosen as the default dialect if the user doesn't acknowledge one of the actual ones. (I'd bet some would argue that UK would be a more correct default, but US is what we develop in.)

If you'd rather wait for a UX improvement to make it more clear that this country selection is just for the language variant, that would also be an option.

**Comment by 13r0ck** [comment]
> There is no "Other" or "None" locale, that's the crux of the issue.
We are in agreement, that was the point I was trying to make.

In michael's comment he stated `In general, None should be automatically converted to US.`
And I was saying generally the same thing as `I'd bet some would argue that UK would be a more correct default`. I guess I looped two conversations into one. But it sounds like we are on the same page.
